### PR TITLE
Support to read FixedFramePose.

### PR DESCRIPTION
**Pull request by zjwoody**
This pr supports to read FixedFramePose data in cartographer_ros.
I'm not sure whether the msgs type of the fixed frame pose is suitable or not. Also the demo_3d.rviz seems have problem now. I want to send this pr to let you review first to check whether it is the correct way to do it. Thanks:)

### CODE REVIEW

File: cartographer_ros/cartographer_ros/trajectory_options.cc [hunk 1]
```diff
@@ -50,6 +50,12 @@ TrajectoryOptions CreateTrajectoryOptions(
   options.provide_odom_frame =
       lua_parameter_dictionary->GetBool("provide_odom_frame");
   options.use_odometry = lua_parameter_dictionary->GetBool("use_odometry");
+  options.fixed_frame_pose_frame =
+      lua_parameter_dictionary->GetString("fixed_frame_pose_frame");
+  options.provide_fixed_frame_pose_frame =
```
Comment (gaschler) [inline cartographer_ros/cartographer_ros/trajectory_options.cc]: This option is not needed, please also remove it in lua configs and so on.
(It appears to be consistent with odometry on first glance, but provide_odom_frame is unrelated, it controls whether TrajectoryState::local_to_map will be published.)

File: cartographer_ros/cartographer_ros/node.cc [hunk 2]
```diff
@@ -52,6 +52,7 @@ cartographer_ros_msgs::SensorTopics DefaultSensorTopics() {
   topics.point_cloud2_topic = kPointCloud2Topic;
   topics.imu_topic = kImuTopic;
   topics.odometry_topic = kOdometryTopic;
+  topics.fixed_frame_pose_topic = kFixedFramePoseTopic;
   return topics;
```
Comment (gaschler) [inline cartographer_ros/cartographer_ros/node.cc]: You need to add a SubscribeWithHandler to subscribers_ in Node::LaunchSubscribers

File: cartographer_ros/cartographer_ros/trajectory_options.cc [hunk 3]
```diff
@@ -50,6 +50,12 @@ TrajectoryOptions CreateTrajectoryOptions(
   options.provide_odom_frame =
       lua_parameter_dictionary->GetBool("provide_odom_frame");
   options.use_odometry = lua_parameter_dictionary->GetBool("use_odometry");
+  options.fixed_frame_pose_frame =
```
Comment (gaschler) [inline cartographer_ros/cartographer_ros/trajectory_options.cc]: I don't think this should be an option. Analogous to IMU, there is an input topic "fixed_frame_pose". We can re-map in the launch file if a different name is needed.

File: cartographer_ros/configuration_files/revo_lds.lua [hunk 4]
```diff
@@ -24,6 +24,9 @@ options = {
   odom_frame = "odom",
   provide_odom_frame = true,
   use_odometry = false,
+  fixed_frame_pose_frame = "fixed_frame_pose",
+  provide_fixed_frame_pose_frame = true,
+  use_fixed_frame_pose = false,
```
Comment (gaschler) [inline cartographer_ros/configuration_files/revo_lds.lua]: If there are new options, defaults also need to be set in https://github.com/googlecartographer/cartographer_toyota_hsr/tree/master/cartographer_toyota_hsr/    https://github.com/googlecartographer/cartographer_turtlebot   https://github.com/googlecartographer/cartographer_fetch/tree/master/cartographer_fetch/configuration_files in follow-up PRs.

File: cartographer_ros/cartographer_ros/node.cc (hunk 2 above)
Comment (zjwoody) [reply cartographer_ros/cartographer_ros/node.cc]: Done. Thanks!

File: cartographer_ros/configuration_files/demo_2d.rviz [hunk 5]
```diff
@@ -77,6 +77,8 @@ Visualization Manager:
           Value: true
         odom:
           Value: true
+        pose_link:
+	  Value: true
```
[the remark this answers is not in the document]
Comment (zjwoody) [reply cartographer_ros/configuration_files/demo_2d.rviz]: Done.

File: cartographer_ros/configuration_files/demo_3d.rviz [hunk 6]
```diff
@@ -79,6 +79,8 @@ Visualization Manager:
           Value: true
         odom:
           Value: true
+        pose_link:
```
[the remark this answers is not in the document]
Comment (zjwoody) [reply cartographer_ros/configuration_files/demo_3d.rviz]: Done.

File: cartographer_ros/configuration_files/demo_2d.rviz [hunk 7]
```diff
@@ -77,6 +77,8 @@ Visualization Manager:
           Value: true
         odom:
           Value: true
+        pose_link:
```
[the remark this answers is not in the document]
Comment (zjwoody) [reply cartographer_ros/configuration_files/demo_2d.rviz]: Done.

File: cartographer_ros/cartographer_ros/trajectory_options.cc (hunk 3 above)
Comment (zjwoody) [reply cartographer_ros/cartographer_ros/trajectory_options.cc]: Done.

File: cartographer_ros/cartographer_ros/trajectory_options.cc (hunk 1 above)
Comment (zjwoody) [reply cartographer_ros/cartographer_ros/trajectory_options.cc]: Done.

**Comment by zjwoody** [comment]
I'll send you the test data later. Thanks!

**Comment by cschuet** [comment]
@zjwoody Sorry this PR got messed up by our merge bot. Could you reupload this?

**Pull request closed without merging**